modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/PrepareServiceImpl.java

```diff
@@ -102,9 +102,14 @@ public PrepareServiceImpl(GridKernalContext ctx) {
                         "querySql=\"" + ctx.query() + "\"]", IgniteQueryErrorCode.UNSUPPORTED_OPERATION);
             }
         }
-        catch (ValidationException | CalciteContextException e) {
+        catch (CalciteContextException e) {
             throw new IgniteSQLException("Failed to validate query. " + e.getMessage(), IgniteQueryErrorCode.PARSING, e);
         }
+        catch (ValidationException e) {
+            String cause = e.getCause() == null ? e.getMessage() : e.getCause().getMessage();
+
+            throw new IgniteSQLException("Failed to validate query. " + cause, IgniteQueryErrorCode.PARSING, e);
+        }
```

Comment:
Let's change `IgnitePlanner.validate()` instead (construct `ValidationException` with the correct message).

Reply:
imho, there is no need to have ValidationException at all. I have killed off it from the code.

Reply:
Done